src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -84,6 +84,29 @@ bool CImageLoaderTGA::isALoadableFileFormat(io::IReadFile* _file) const
 	return true;
 }
 
+// convertColorFlip() does color conversion as well as taking care of properly flipping the given image.
+template <typename T, E_FORMAT srcFormat, E_FORMAT destFormat>
+static void convertColorFlip(const T *src, T *dest, core::vector3d<uint32_t> &size, int c, bool flip)
+{
+	T *in = (T *) src;
+	T *out = (T *) dest;
+	uint32_t stride = size.X * c;
```

Comment:
your strides will be different for non-identical src and dest Formats

moreover images usually have an issue with row-pitch alignment (for cache, processor architecture reasons 8 byte alignment works best, so you cannot really have 8bit images of arbitrary width, they get a padding)

Reply:
just pass src and dest row pitches, instead of `c`

Reply:
@devshgraphicsprogramming @Florastamine: is this solved?

Reply:
no idea where that function is now

Reply:
@Florastamine?

Reply:
It's solved - I fixed the stride as instructed.
https://github.com/buildaworldnet/IrrlichtBAW/blob/e28d94463b1e9b758b83a50a79c77ac6cb329677/src/irr/asset/CImageLoaderTGA.cpp#L121

Reply:
still not fixed, using same stride for both the `in` and `out` pointer

Reply:
@Florastamine

Reply:
`src` is a raw pixel buffer loaded via libjpeg, libpng or `IReadFile::read()`. `convertColorFlip()`'s job is just filling a `CImageData` with that raw buffer, doing color conversion in between.

It doesn't do `CImageData`-to-`CImageData` type of conversion, hence only 1 stride.

Reply:
plus if you look at `CImageLoaderTGA.cpp`, dest and src formats on where `convertColorFlip<>()` is used is almost the same, because it's mostly used for flipping images/shuffling RGBA order

Reply:
> It doesn't do CImageData-to-CImageData type of conversion, hence only 1 stride

If you expect those kind of things, then you need to place `assert`s in the code